src/Serilog.Extensions.Hosting/Extensions/Hosting/DiagnosticContextCollector.cs

```diff
@@ -38,19 +39,59 @@ public void AddOrUpdate(LogEventProperty property)
             }
         }
 
+        /// <summary>
+        /// Set an exception to include in the Serilog request log event.
+        /// </summary>
+        /// <example>
+        /// Passing an exception to the diagnostic context is useful when unhandled exceptions are handled before reaching Serilog's
+        /// RequestLoggingMiddleware. One example is using https://www.nuget.org/packages/Hellang.Middleware.ProblemDetails to transform
+        /// exceptions to ProblemDetails responses.
+        /// </example>
+        /// <remarks>
+        /// If an unhandled exception reaches Serilog's RequestLoggingMiddleware, then the unhandled exception takes precedence.<br/>
+        /// If <c>null</c> is given, it clears any previously assigned exception.
+        /// </remarks>
+        /// <param name="exception">The exception to log.</param>
+        public void SetException(Exception exception)
+        {
+            lock (_propertiesLock)
+            {
+                if (_properties == null) return;
+                _exception = exception;
+            }
+        }
+
         /// <summary>
         /// Complete the context and retrieve the properties added to it, if any. This will
         /// stop collection and remove the collector from the original execution context and
         /// any of its children.
         /// </summary>
         /// <param name="properties">The collected properties, or null if no collection is active.</param>
         /// <returns>True if properties could be collected.</returns>
+        /// <seealso cref="IDiagnosticContext.Set"/>
         public bool TryComplete(out IEnumerable<LogEventProperty> properties)
```

Comment:
Perhaps this overload could be marked obsolete? Removing it would break binary compat with older _Serilog.AspNetCore_ assemblies, but new code should really be using the new overload...